x/incentive/migrations/v2/store.go

```diff
@@ -0,0 +1,35 @@
+package v2
+
+import (
+	"context"
+
+	bbntypes "github.com/babylonlabs-io/babylon/v3/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/babylonlabs-io/babylon/v3/x/incentive/types"
+)
+
+// Keeper the expected keeper interface to perform the migration
+type Keeper interface {
+	SetFinalityProviderCurrentRewards(ctx context.Context, fp sdk.AccAddress, rwd types.FinalityProviderCurrentRewards) error
+	IterateFpCurrentRewards(ctx context.Context, it func(fp sdk.AccAddress, fpCurrRwds types.FinalityProviderCurrentRewards) error) error
+}
+
+// MigrateStore performs in-place store migrations.
+// Migration updates all finality provider current rewards by the hardcoded decimal rewards
+// which adds 20 decimals points to increase precision at the time to calculate how much
+// rewards each satoshi staked is entitled to receive.
+func MigrateStore(
+	ctx sdk.Context,
+	k Keeper,
+) error {
+	return k.IterateFpCurrentRewards(ctx, func(fp sdk.AccAddress, fpCurrRwds types.FinalityProviderCurrentRewards) error {
```

Comment:
It would be good to do an experiment with lots of rewards how long migration will take (this can be done after merging pr)

Reply:
it should be a small number, the number of different active fps that got rewards at some point. But I will ask devops to get this data from mainnet so we can make sure it will not take long